terraform/modules/matrix-service/driver_lambda.tf

```diff
@@ -0,0 +1,85 @@
+resource "aws_iam_role" "matrix_service_driver_lambda" {
+  name = "matrix-service-driver-daemon-${var.deployment_stage}"
+
+  assume_role_policy = <<POLICY
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+POLICY
+}
+
+resource "aws_iam_role_policy" "matrix_service_driver_lambda" {
+  name = "matrix-service-driver-daemon-${var.deployment_stage}"
+  role = "${aws_iam_role.matrix_service_driver_lambda.name}"
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "LogsPolicy",
+      "Effect": "Allow",
+      "Resource": "arn:aws:logs:*:*:*",
+      "Action": [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ]
+    },
+    {
+      "Sid": "DynamoPolicy",
+      "Effect": "Allow",
+      "Action": [
+        "dynamodb:PutItem"
+      ],
+      "Resource": "arn:aws:dynamodb:us-east-1:861229788715:table/dcp-matrix-service-state-table-${var.deployment_stage}"
+    },
+    {
+      "Sid": "LambdaPolicy",
+      "Effect": "Allow",
+      "Action": [
+        "lambda:InvokeFunction"
+      ],
+      "Resource": "arn:aws:lambda:us-east-1:861229788715:function:dcp-matrix-service-mapper-${var.deployment_stage}"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_s3_bucket" "matrix_service_lambda_deployment_bucket" {
+    bucket = "dcp-matrix-service-lambda-deployment-${var.deployment_stage}"
+    acl = "private"
+    force_destroy = "false"
+    acceleration_status = "Enabled"
+}
+
+resource "aws_lambda_function" "matrix_service_driver_lambda" {
+  function_name    = "dcp-matrix-service-driver-${var.deployment_stage}"
+  s3_bucket        = "${aws_s3_bucket.matrix_service_lambda_deployment_bucket.id}"
+  s3_key           = "driver_daemon.zip"
+  role             = "${aws_iam_role.matrix_service_driver_lambda.arn}"
+  handler          = "app.driver_handler"
+  runtime          = "python3.6"
+  timeout          = 300
+
+  environment {
+    variables = {
+        DEPLOYMENT_STAGE = "${var.deployment_stage}"
+        LAMBDA_DRIVER_FUNCTION_NAME="dcp-matrix-service-driver-${var.deployment_stage}"
```

Comment:
why does the driver need all this lambda variables. Doesn't it only need to know about the next one in the sequence?

Reply:
Hm yea that makes sense. Problem is importing https://github.com/HumanCellAtlas/matrix-service/pull/52/files#diff-6d63faf8105e81f612d9bfee6d1d5aefR8 requires all 4 to be set. I personally think enums are a nice plus for programmatically organizing keys. Thoughts?

Reply:
Ah I see. I agree the enum is nice. You could just change the code in the link to utilize os.getenv. That way any variables not appropriate for a lambda will get set to None and fail on invocation attempt of a lambda it should not have permission to invoke.